core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -1595,6 +1598,17 @@ protected static Calendar getCalendarNotTooNear(int timeUnit) {
     tester.checkFails(
         "cast(cast('blah' as varchar(10)) as boolean)", INVALID_CHAR_MESSAGE,
         true);
+
+    tester.checkBoolean("cast(0 as boolean)", false);
```

Comment:
For consistency, it should be `Boolean.FALSE` instead of `false`

Reply:
got it